nodeadm/internal/aws/imds/imds.go

```diff
@@ -3,16 +3,39 @@ package imds
 import (
 	"context"
 	"io"
+	"net/http"
+	"net/url"
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	awshttp "github.com/aws/aws-sdk-go-v2/aws/transport/http"
 	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 )
 
 var Client *imds.Client
 
+// This function is a wrapper around the default `http.ProxyFromEnvironment` function
+// which cachces the proxy variables in environment when first invoked thus, preventing subsequent
+// configuration attempts of http proxy via derived from user-data. Since, the first outbound
+// API call in nodeadm is to the IMDS for fetching user-data, we bypass caching.
```

Comment:
seems like it might be fragile, wouldn't this break if we added another outbound HTTP call that didn't go to localhost?